src/sentry/tasks/store.py

```diff
@@ -582,6 +583,16 @@ def _do_save_event(
             raise
 
         finally:
+            if (
+                consumer_type == ConsumerType.Transactions
+                and event_id
+                and in_rollout_group("transactions.do_post_process_in_save", event_id)
+            ):
+                # we won't use the transaction data in post_process
+                # so we can delete it from the cache now.
+                if cache_key:
+                    processing_store.delete_by_key(cache_key)
+
```

Comment:
just double checking - should this be in the `finally` block, or in an `else` block?

do we want to run it if if we hit the save event exception above?

Reply:
i think we'd always want to do the processing store deletion -- otherwise we risk filling up the processing store in the case where we're not running transactions through post process